test/test_ops.py

```diff
@@ -0,0 +1,248 @@
+
+
+@given(x=tensor_clamp_input())
+def test_result_of_tensor_clamp(x):
+    minimum = torch.tensor(x[0])
+    value = torch.tensor(x[1])
+    maximum = torch.tensor(x[2])
+
+    output = tensor_clamp(value, minimum, maximum)
+    expected_output = []
+    for i in range(minimum.size()[0]):
+        expected_output.append(torch.clamp(value[i], x[0][i], x[2][i]))
+    expected_output = torch.tensor(expected_output)
+
+    assert ((output >= minimum).all() and (output <= maximum).all())
+    assert (torch.allclose(expected_output, output, RTOL, ATOL))
+
+
+@given(x=tensor_clamp_ste_input())
+def test_ste_of_tensor_clamp_ste(x):
+    minimum = torch.tensor(x[0])
+    value = torch.tensor(x[1], requires_grad=True)
+    grad = x[2]
+    grad = torch.tensor(grad)
+    maximum = torch.tensor(x[3])
+
+    output = tensor_clamp_ste(value, minimum, maximum)
+
+    output.backward(grad, retain_graph=True)
+    assert (torch.allclose(grad, value.grad, RTOL, ATOL))
+
+
+# Test different combinations of Narrow Range (True/False) and BitWidth (1...8)
+@given(narrow_range=st.booleans(), bit_width=st.integers(min_value=1, max_value=8))
+def test_result_of_max_uint(narrow_range, bit_width):
+    bit_width = torch.tensor(bit_width, dtype=torch.float)
+    output = max_uint(narrow_range, bit_width)
+
+    if narrow_range:
+        expected_output = (2 ** bit_width) - 2
+    else:
+        expected_output = (2 ** bit_width) - 1
+    expected_output = expected_output
+
+    assert (torch.allclose(expected_output, output, RTOL, ATOL))
+
+
+# Test different combinations of Narrow Range (True/False) and BitWidth (1...8)
+@given(signed=st.booleans(), bit_width=st.integers(min_value=1, max_value=8))
+def test_result_of_max_int(signed, bit_width):
+    bit_width = torch.tensor(bit_width, dtype=torch.float)
+    output = max_int(signed, bit_width)
+
+    if signed:
+        expected_output = (2 ** (bit_width - 1)) - 1
+    else:
+        expected_output = (2 ** bit_width) - 1
+    expected_output = expected_output
+
+    assert (torch.allclose(expected_output, output, RTOL, ATOL))
+
+
+# Test different combinations of Narrow Range (True/False), Signed (True/False), and BitWidth (1...8)
+@given(narrow_range=st.booleans(), signed=st.booleans(), bit_width=st.integers(min_value=0, max_value=8))
+def test_result_of_min_int(narrow_range, signed, bit_width):
+    bit_width = torch.tensor(bit_width, dtype=torch.float)
+    output = min_int(signed, narrow_range, bit_width)
+
+    if signed and narrow_range:
+        expected_output = -(2 ** (bit_width - 1)) + 1
+    elif signed and not narrow_range:
+        expected_output = -(2 ** (bit_width - 1))
+    else:
+        expected_output = torch.tensor(0.0)
+
+    expected_output = expected_output
+
+    assert (torch.allclose(expected_output, output, RTOL, ATOL))
+
+
+# Requires two floats as maximum and minimum and a tensor of float
+@given(minmax=two_ordered_numbers(), x=list_float_st)
```

Comment:
Are we sure this test the cases where there some clamping happening?

Reply:
Leaving this as memo. The only constraint set is that min_value < max_value, while there are no constraints on the value in x. Considering that the number are randomly generated, there is high chance that the clamp operation is really executed. A similar thing can be said and extended for the tensor_clamp case.